client/boot/index.js

```diff
@@ -361,6 +362,20 @@ function reduxStoreReady( reduxStore ) {
 		require( 'lib/rubberband-scroll-disable' )( document.body );
 	}
 
+	/*
+	 * Layouts with differing React mount-points will not reconcile correctly,
+	 * so remove an existing single-tree layout by re-rendering if necessary.
+	 *
+	 * TODO (@seear): React 15's new reconciliation algo may make this unnecessary
+	 */
+	page( '*', function( context, next ) {
+		if ( ! isIsomorphicRoute( context.path ) && previousLayoutIsSingleTree() ) {
```

Comment:
Back in #4864, with /start as layout-last and the necessary unmounts for that, `! isIsomorphicRoute()` was necessary to prevent an infinite loop where the two unmounts battled each other.

It is not necessary for that now, but if we removed it, we would get this re-render on any transition from layout-last route to layout-last route, e.g. logged-out /design to /theme.

Maybe we should take that hit to keep things simpler and hope React 15 solves it?